functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,276 @@ export const processNewAllotment = functions.database
+                ending: moment.duration(row['Ending']).asSeconds(),
+                continuationFrom,
+                contentReporting: {
+                    date: row['Date'],
+                    locatioon: row['Location'],
+                    category: row['Category'],
+                    topics: row['Topics'],
+                    gurudevaTimings: row['Gurudeva Timings'],
+                    otherSpeakers: row['Other Speakers'],
+                    kirtan: row['Kirtan'],
+                    abruptLecture: row['Abrupt Lecture'],
+                    suggestedTitle: row['Suggested Title'],
+                    languages: row['Languages'],
+                    soundQualityRating: row['Sound Quality'],
+                    soundIssues: row['Sound Issues'],
+                    comments: row['Comments'],
+                    submissionTimestamp: row['Timestamp'],
+                    submissionSerial: row['Submission Serial'],
+                },
+                importTimestamp: admin.database.ServerValue.TIMESTAMP,
+                processingResolution: row['Resolution'],
+                _import
+            };
+
+
+            
+            // Same file name --> add newly read chunk (row) to the `currentFile` object
+            if (currentFile.file_name === AudioFileName) 
+                currentFile.chunks.push(chunkToStore);
+            
+            // NEW FILE --> Save old file AND add the newly read chunk to `currentFile` object            
+            else {
+                if (currentFile.file_name != null) {
+
+                    // if file object exists, add the chunks to it
+                    if (chunks[currentFile.file_name]) 
+                        chunks[currentFile.file_name].push(currentFile.chunks)
+
+                    // if this is the first occurence of the file, set its value to the list of chunks 
+                    else    
+                        chunks[currentFile.file_name] = currentFile.chunks;
+                }
+
+                currentFile.file_name = AudioFileName;
+                currentFile.chunks = [chunkToStore];
+            }                    
+        }
+        
+
+        ////////////////////////////////
+        // Processing Sheet Audio Files
+        ////////////////////////////////
+
+        Object.keys(chunks).forEach(async fileName => {
+            let currentFilechunks = chunks[fileName];
+
+            // Sorting by `beginning` time
+            currentFilechunks = await lodash.orderBy(currentFilechunks, ['beginning'], ['asc']);
+            // console.log(currentFilechunks);
+
+            let chunksToImport = [];
+            ////////////////////////////////////////////
+            // [CHECK #5] Chunks of a single audio file 
+            // should not overlap each other in timing.
+            ////////////////////////////////////////////
+            let lastEndingTime = null;
+            currentFilechunks.forEach((chunk, i) => {
```

Comment:
Validating the whole file with all its chunks should be extracted to a separate function.

As I see now, we have per-row validation rules and per-file. The latter ones are spanning all the chunks. These are: overlapping chunks and `continuationFrom`. So have two functions: `validateSpreadsheetRow` and `validateFileChunks`.